# Week 1 - Research Question

## 2. Research Question

### Search Question

Which webpages should be prioritized for content refresh based on their current performance and engagement signals?

### Unit of Analysis

One webpage (one row in the dataset represents one content page).

### Output

A priority score indicating whether a webpage should be reviewed for content refresh.

### Decision

The SEO/content team decides which pages should be refreshed first.

### Action

Pages with the highest opportunity scores are added to the content review queue before lower-priority pages.

### Cost of a Wrong Recommendation

If an important page is not selected, the organization may lose search traffic and potential customers.

If a low-value page is selected unnecessarily, time and editorial resources may be wasted.

### Why Machine Learning?

The dataset contains many signals, including search volume, CTR, ranking position, engagement, content age, and trend information. Evaluating all these signals manually across thousands of pages is difficult. Machine learning can identify useful patterns and support better prioritization decisions.

In [15]:
import pandas as pd

In [16]:
df = pd.read_csv("content_refresh_anonymized.csv")

In [17]:
df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [18]:
print("Rows : ", len(df))

Rows :  30000


In [19]:
print("Columns : ", len(df.columns))

Columns :  44


In [20]:
print("Average CTR : ", round(df["ctr"].mean(),2))

Average CTR :  0.51


In [21]:
print("Average Position : ",round(df["avg_position"].mean(),2))

Average Position :  16.34


In [22]:
print("Down Trend Pages : ", (df["trend_direction"]=="down").sum())

Down Trend Pages :  16262


## 3. Quick Look at the Data

The starter dataset contains **30000** webpages and **44** features.

The average click-through rate (CTR) is **0.51**, while the average search position is **16.34**.

A total of **16262** pages are currently marked as **downward trending**, suggesting that content prioritization could help identify pages that deserve review.

These observations indicate that the dataset contains enough variation to support a data-driven prioritization approach.

## 4. Careful Words

This notebook presents an exploratory framing of the problem.

The observations describe patterns in the available dataset and should not be interpreted as evidence of causation.

Any future recommendation model should be treated as a decision-support tool rather than an automatic decision-maker.

Further validation will be required before deploying such recommendations in production.

## 5. Self Check

- [x] Selected a predefined lane
- [x] Defined the research question
- [x] Explained the decision and action
- [x] Identified the unit of analysis
- [x] Included real statistics from the dataset
- [x] Explained why ML can help
- [x] Used careful, non-causal language
- [x] Focused on decision support instead of just training a model